src/platform/packages/private/kbn-index-editor/src/index_update_service.ts

```diff
@@ -176,6 +194,17 @@ export class IndexUpdateService {
   public readonly dataView$: Observable<DataView> = combineLatest([
     this._indexName$,
     this._indexCrated$,
+    this.pendingColumnsToBeSaved$.pipe(
+      startWith([]),
+      // Refetch the dataView to look for new field types when there are new columns saved
+      // (when pendingColumnsToBeSaved$ length decreases)
+      scan((prev, curr) => ({ prevLength: prev.currLength ?? 0, currLength: curr.length }), {
+        prevLength: 0,
+        currLength: 0,
+      }),
+      filter(({ prevLength, currLength }) => currLength < prevLength || prevLength === 0),
+      startWith({ prevLength: 0, currLength: 0 })
+    ),
```

Comment:
This logic seems overcomplicated to me. Can we simply refetch on manual refresh after saving? You can bump the refreshSubject on save